Src/FluentAssertions/Equivalency/SelfReferenceEquivalencyAssertionOptions.cs

```diff
@@ -757,24 +716,7 @@ public override string ToString()
             .AppendLine("- Compare tuples by their properties")
             .AppendLine("- Compare anonymous types by their properties");
 
-        if (compareRecordsByValue is true)
-        {
-            builder.AppendLine("- Compare records by value");
-        }
-        else
-        {
-            builder.AppendLine("- Compare records by their members");
-        }
-
-        foreach (Type valueType in valueTypes)
-        {
-            builder.AppendLine(CultureInfo.InvariantCulture, $"- Compare {valueType} by value");
-        }
-
-        foreach (Type type in referenceTypes)
-        {
-            builder.AppendLine(CultureInfo.InvariantCulture, $"- Compare {type} by its members");
-        }
+        equalityStrategyCache.AppendToString(builder);
```

Comment:
Would it be prettier to expose `IEnumerable<Type> ValueTypes` and `IEnumerable<Type> ReferenceTypes` instead of `EqualityStrategyCache` knowing about the `StringBuilder`?

I'm good with either since this is internal stuff.

Reply:
I have no direct preference.
I also thought about adding a normal `ToString()` method instead and use the string from there instead of using a StringBuilder, but this could impact performance (but I doubt in a meaningful way)...

Reply:
I changed it to use the `ToString()` overload instead.